spec/new_integration/dotenv_loading_spec.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/FetchEnvVar
+
+RSpec.describe "Dotenv loading", :app_integration do
+  before do
+    @orig_env = ENV.to_h
+  end
+
+  after do
+    ENV.replace(@orig_env)
+  end
+
+  context "dotenv gem is available" do
+    before do
+      require "dotenv"
+    end
+
+    context "hanami env is development" do
+      it "loads .env.development.local, .env.local, .env.development and .env (in this order) into ENV", :aggregate_failures do
```

Comment:
FWIW, a lot of these tests now overlap with the ones I added in the (also open) https://github.com/hanami/hanami/pull/1186. Once these are both merged, I'll remove any redundant tests (leaving this file as the canonical set of tests exercising the `.env` file loading behaviour).

In the meantime, I figured it made sense to let each PR stand on its own.